DisguiseUnityRenderStream/Runtime/ScratchTextureManager.cs

```diff
@@ -0,0 +1,132 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Disguise.RenderStream
+{
+    /// <summary>
+    /// A simpler version of <see cref="Disguise.RenderStream.TemporaryTexture2DManager"/> that doesn't need to manage texture lifetime.
+    /// </summary>
+    public abstract class ScratchTextureManager<TTexture>
+    {
+        public bool DebugTrace { get; set; }
+        
+        readonly Dictionary<Texture2DDescriptor, TTexture> m_Items = new Dictionary<Texture2DDescriptor, TTexture>();
+
+        protected string m_Name;
+        
+        public void Clear()
+        {
+            DebugLog(DebugTrace, $"Cleared textures");
+            
+            foreach (var texture in m_Items.Values)
+            {
+                DestroyTexture(texture);
+            }
+            
+            m_Items.Clear();
+        }
+        
+        public TTexture Get(Texture2DDescriptor descriptor)
+        {
+            if (m_Items.TryGetValue(descriptor, out var item))
+            {
+                return item;
+            }
+            else
+            {
+                DebugLog(DebugTrace, $"Created texture: {descriptor}");
+                
+                var texture = CreateTexture(descriptor);
+                m_Items.Add(descriptor, texture);
+                return texture;
+            }
+        }
+
+        protected abstract TTexture CreateTexture(Texture2DDescriptor descriptor);
+        
+        protected abstract void DestroyTexture(TTexture texture);
+        
+        void DebugLog(bool condition, string message)
+        {
+            if (condition)
```

Comment:
Do we need the `condition` param? You're just using the `DebugTrace` property, right?